qiskit/transpiler/target.py

```diff
@@ -590,6 +590,245 @@ def operation_names_for_qargs(self, qargs):
+                if None in self._gate_map[op_name]:
+                    out_dict[op_name] = self._gate_map[op_name][None]
+                if self._gate_map[op_name] is None:
+                    out_dict[op_name] = None
+        if not out_dict:
+            raise KeyError(
+                f"Operation class {operation_class} on qargs: {qargs} with parameters "
+                f"{parameters} is not supported"
+            )
+        return out_dict
+
+    def get_instruction_errors_by_class(self, operation_class, qargs, parameters=None):
+        """Return the error rates for instructions by operation class
+
+        This method and looking up by class is in general more expensive to compute as it
+        needs to iterate over all operations in the target instead of just a single lookup
+        by name as is done in :meth:`~.get_instruction_error`. This method can also
+        return multiple properties as there might be multiple implementations of a class in the
+        target. The typical use case for this operation is to get the properties of a specific
+        variant of an operation on the backend. For example, if you wanted to get the properties
+        of all the instructions capable of implementing a :class:`~.RXGate` on a specific qubit
+        with a fixed angle.
+
+        Args:
+            operation_class (qiskit.circuit.Operation): The operation class to lookup the
+                properties for
+            qargs (tuple): The tuple of qubit indices for the instruction.
+            parameters (list): A list of parameters to filter the returned error rates on
+                whether an operation supports them on the specified qubits. If the instruction
+                supports the parameter values specified in the list on the operation and qargs
+                specified the properties for that instruction will be included in the return.
+                If this argument is not specified this method will return the properties for
+                any instruction that is supported independent of the instruction parameters. If
+                specified with any :class:`~.Parameter` objects in the list, that entry will be
+                treated as supporting any value, however parameter names will not be checked (for
+                example if an operation in the target is listed as parameterized with ``"theta"``
+                and ``"phi"`` is passed into this function that will return the error rate for
+                that instruction). For example, if called with::
+
+                    parameters = [Parameter("theta")]
+                    target.get_errors_by_class(RXGate, (0,), parameters=parameters)
+
+                will return the properties of any :class:`~.RXGate` instruction that is suporrted on
+                qubit 0 which will accept any parameter (excluding fixed angle variants).
+
+        Returns:
+            dict: A dictionary mapping all the operation names to error rates for all matching
```

Comment:
I'm not sure how this method will be used in practice. Even if we get multiple errors for different gate implementations, transpiler passes don't choose most efficient _pulse gate_ , i.e. `.add_calibration`, so fidelity metric of optimized sequence doesn't match. I think we can drop these methods if we don't have practical use case. User can still get the same function by creating the reverse mapping of `Target._gate_name_map`.